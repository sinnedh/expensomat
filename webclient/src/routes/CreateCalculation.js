import React from 'react';
import { Redirect } from 'react-router-dom';
import { createCalculation } from '../actions';
import CalculationForm from '../components/CalculationForm';

class CreateCalculation extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      redirect_to: '',
    }
  }

  createCalculation = (calculation) => {
    this.props.dispatch(createCalculation(calculation));
  }

  render() {
    return (
      <div>
        {this.state.redirect_to !== '' && <Redirect to={this.state.redirect_to} />}
        <h1>Create new calculation</h1>
        <CalculationForm handleSubmit={this.createCalculation}/>
      </div>
    );
  }
}

export default CreateCalculation;
