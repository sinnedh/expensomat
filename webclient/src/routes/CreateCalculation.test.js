import React from 'react'
import { Map } from 'immutable'
import configureStore from 'redux-mock-store'
import CreateCalculation from './CreateCalculation'

describe('CreateCalculation', () => {
  it('renders properly', () => {
    const initialState = Map()
    const mockStore = configureStore()
    const store = mockStore(initialState)
    const wrapper = shallow(<CreateCalculation store={store} />)
    expect(wrapper.dive()).toMatchSnapshot()
  })


//  it('calls cC properly', () => {
//    const initialState = {}
//    const mockStore = configureStore()
//    const store = mockStore(initialState)
//    const wrapper = mount(<CreateCalculation store={store} />)
//    wrapper.props.createCalculation = jest.fn()
//    //wrapper.props.createCalculation({})
//    wrapper.find('input.submit').simulate('click', { preventDefault() {} })
//    expect(dispatch).toBeCalled()
//
//  })

})
