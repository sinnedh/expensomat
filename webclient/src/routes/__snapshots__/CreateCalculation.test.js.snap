// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCalculation renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateCalculation
    createCalculation={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <h1>
          Create new calculation
</h1>,
        <CalculationForm
          handleSubmit={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Create new calculation",
        },
        "ref": null,
        "rendered": "Create new calculation",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <h1>
            Create new calculation
</h1>,
          <CalculationForm
            handleSubmit={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Create new calculation",
          },
          "ref": null,
          "rendered": "Create new calculation",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
